dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -211,3 +232,110 @@ func validateFromChainState(runtimeInstance parachainruntime.RuntimeInstance, po
 
 	return &candidateCommitments, persistedValidationData, isValid, nil
 }
+
+// validateFromExhaustive validates a candidate parachain block with provided parameters
+func validateFromExhaustive(validationHost parachainruntime.ValidationHost,
+	persistedValidationData parachaintypes.PersistedValidationData,
+	validationCode parachaintypes.ValidationCode,
+	candidateReceipt parachaintypes.CandidateReceipt, pov parachaintypes.PoV) (
+	*ValidationResult, error) {
+
+	validationCodeHash := validationCode.Hash()
+	// basic checks
+	validationErr, internalErr := performBasicChecks(&candidateReceipt.Descriptor, persistedValidationData.MaxPovSize,
+		pov,
+		validationCodeHash)
+	if internalErr != nil {
+		return nil, fmt.Errorf("performing basic checks: %w", internalErr)
+	}
+
+	if validationErr != nil {
+		validationResult := &ValidationResult{
+			InvalidResult: validationErr,
+		}
+		return validationResult, nil //nolint: nilerr
+	}
+
+	validationParams := parachainruntime.ValidationParameters{
+		ParentHeadData:         persistedValidationData.ParentHead,
+		BlockData:              pov.BlockData,
+		RelayParentNumber:      persistedValidationData.RelayParentNumber,
+		RelayParentStorageRoot: persistedValidationData.RelayParentStorageRoot,
+	}
+
+	validationResult, err := validationHost.ValidateBlock(validationParams)
+	if err != nil {
+		return nil, fmt.Errorf("executing validate_block: %w", err)
+	}
+
+	validationResultHash, err := validationResult.HeadData.Hash()
+	if err != nil {
+		return nil, fmt.Errorf("hashing validation result: %w", err)
```

Comment:
rename `validationResultHash` to `headDataHash`. Also, change the error string.

```suggestion
		return nil, fmt.Errorf("hashing head data: %w", err)
```

Reply:
updated.

Reply:
You have missed updating the variable from `validationResultHash` to `headDataHash`.

Reply:
Updated.